consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/support/StubbedSynchronizerUpdater.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.consensus.ibft.support;
+
+import static java.util.function.Function.identity;
+
+import tech.pegasys.pantheon.consensus.ibft.SynchronizerUpdater;
+import tech.pegasys.pantheon.ethereum.p2p.api.PeerConnection;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class StubbedSynchronizerUpdater implements SynchronizerUpdater {
+
+  private Map<PeerConnection, ValidatorPeer> validatorNodes;
+
+  @Override
+  public void updatePeerChainState(
+      final long knownBlockNumber, final PeerConnection peerConnection) {
+    validatorNodes.get(peerConnection).updateEstimatedChainHeight(knownBlockNumber);
+  }
+
+  public void addNetworkPeers(final Collection<ValidatorPeer> nodes) {
```

Comment:
Is this an `add` operations or a `set` operation? ...what happens when its invoked twice?

Reply:
any thoughts? @rain-on

Reply:
just saw this - create the map instantiation, then putAll in here. Thus will continue to grow as you call "add" - given its called once in the integration test framework setup, should be acceptable (i.e. doesn't require a "clear()" call atm.